src/Symfony/Component/Translation/Parameter/MoneyParameter.php

```diff
@@ -0,0 +1,59 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Translation\Parameter;
+
+use Money\Currencies\ISOCurrencies;
+use Money\Formatter\DecimalMoneyFormatter;
+use Money\Money;
+use Symfony\Contracts\Translation\ParameterInterface;
+
+/**
+ * Wrapper around PHP NumberFormatter for money
+ * The provided currency is used instead of the locale's currency.
+ *
+ * @author Sylvain Fabre <syl.fabre@gmail.com>
+ */
+class MoneyParameter implements ParameterInterface
+{
+    private $value;
+    private $currency;
+    private $style;
+
+    private $formatters = [];
+
+    public function __construct($value, string $currency, int $style = \NumberFormatter::CURRENCY)
+    {
+        $this->value = $value;
+        $this->currency = $currency;
+        $this->style = $style;
+    }
+
+    public function format(string $locale = null): string
+    {
+        if (!isset($this->formatters[$locale][$this->style])) {
```

Comment:
Same here.

Reply:
I think someone may want to use `NumberFormatter::CURRENCY_ACCOUNTING`.

But it may be implemented with another `AccountingParameter`

Reply:
The way the code is written, it MUST be another `MoneyParameter` object. 

Im not saying that you should remove the `$style` property. Im suggesting you to rewrite this line to: 

```suggestion
        if (!isset($this->formatters[$locale]) {
```

(And fix other lines with the same change.

Reply:
Ok got it 🙈